crux/models/resource.py

```diff
@@ -307,35 +236,28 @@ def update(self, name=None, description=None, tags=None, provenance=None):
         body = {}  # type: Dict[str, Union[str, List, Dict]]
 
         if name is not None:
-            body["name"] = name
+            self.object_model["name"] = name
         if description is not None:
-            body["description"] = description
+            self.object_model["description"] = description
         if tags is not None:
-            if isinstance(tags, list):
-                body["tags"] = tags
-            else:
-                raise TypeError("Tags should be of type list")
+            self.object_model["tags"] = tags
         if provenance is not None:
-            body["provenance"] = provenance
-
-        if body:
-            response = self.connection.api_call(
-                "PUT", ["resources", self.id], headers=headers, json=body
-            )
-
-            response_dict = response.json()
-
-            if "name" in response_dict:
-                self._name = response_dict["name"]
-            if "tags" in response_dict and tags is not None:
-                self._tags = response_dict["tags"]
-            if "description" in response_dict:
-                self._description = response_dict["description"]
-            if "provenance" in response_dict:
-                self._provenance = response_dict["provenance"]
-            return True
-        else:
-            raise ValueError("Name, Description, Tags or Provenance should be set")
+            self.object_model["provenance"] = provenance
+
+        body = self.to_dict()
+
+        log.debug("Body %s", body)
+
+        resource_object = self.connection.api_call(
+            "PUT", ["resources", self.id], headers=headers, json=body, model=Resource
```

Comment:
Is `self.id` doesn't exist (new object) this will fail. It should probably fail with an explicit error, and force people to use `create()`.